### PR TITLE
[#4836] EventBus impl - Remove duplicate fields and args, rename confusing field and method names

### PR DESCRIPTION
**Pull request by magicprinc**
https://github.com/eclipse-vertx/vert.x/issues/4836

1) duplicate fields as in superclass
2) unregister(): ArrayDeque<>(8)
3) @Override on doReceive
4) javadoc, spelling

Eclipse Contributor Agreement - signed.

PS: second attempt to make right PR

### CODE REVIEW

**Comment by magicprinc** [comment]
@tsegismont @He-Pin Could we manage this PR too? 😅🙏
It is less complication than previous.

File: src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java [hunk 1]
```diff
@@ -65,8 +59,8 @@ void fail(ReplyException failure) {
   private void doFail(ReplyException failure) {
     trace(null, failure);
     result.fail(failure);
-    if (eventBus.metrics != null) {
-      eventBus.metrics.replyFailure(repliedAddress, failure.failureType());
+    if (bus.metrics != null) {
```
Comment (He-Pin) [inline src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java:62]: better keep it named as eventbus

Comment (magicprinc) [reply src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java]: Thank you! Renamed in parent class!

**Comment by magicprinc** [comment]
The code is compact, clear, and very easy to understand now!

Dear maintainers!
Please, take a look! 
The code is "Finger lickin' good": nothing new, just extra fat removed

**Comment by magicprinc** [comment]
New PR is coming: based on master, with single clean commit

**Comment by magicprinc** [comment]
Better version of this PR: https://github.com/eclipse-vertx/vert.x/pull/4859

**Pull request closed without merging**